Orange/preprocess/impute.py

```diff
@@ -26,7 +27,61 @@ def transform(self, c):
         return numpy.where(numpy.isnan(c), self.value, c)
 
 
-class Average:
+class BaseImputeMethod:
+    name = ""
+    short_name = ""
+    description = ""
+    format = "{var.name} -> {short_name}"
+    columns_only = False
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    @abstract
+    def __call__(self, data, variable):
+        raise NotImplementedError()
+
+    def str(self, var):
+        return self.format.format(var=var, short_name=self.short_name)
```

Comment:
This method could use a docstring describing what it is supposed to return and how does that differ from __str__. Depending on the purpose, a more descriptive method name could be used.